pkg/fqdn/rulegen.go

```diff
@@ -274,25 +284,24 @@ func (gen *RuleGen) UpdateGenerateDNS(lookupTime time.Time, updatedDNSIPs map[st
 		log.WithField("uuid", strings.Join(notFoundUUIDs, ",")).
 			Debug("Did not find all rules during update")
 	}
-	generatedRules, namesMissingIPs := gen.GenerateRulesFromSources(rulesToUpdate)
+	generatedRules, namesMissingIPs, selectorIPMapping := gen.GenerateRulesFromSources(rulesToUpdate)
 	if len(namesMissingIPs) != 0 {
 		log.WithField(logfields.DNSName, strings.Join(namesMissingIPs, ",")).
 			Debug("No IPs to insert when generating DNS name selected by ToFQDN rule")
 	}
 
-	// no new rules to add, do not call AddGeneratedRules below
-	if len(generatedRules) == 0 {
```

Comment:
This is a notable change cc: @raybejjani because we want to always update the SelectorCache. Please shout if you think I need to do more work here. I elected to not do so because CIDR rule generation is going to be removed soon anyway.

Reply:
I don't remember why this ended up like this. It probably was to avoid waiting on locks since PolicyAdd no-oped on no rules (I think). I think this is totally fine. We can always guard against it elsewhere, which may be better anyway.

An alternative would have been to maintain two callbacks, one for selectors that is called unconditionally, and another for the older datapath with the length logic. We're probably too far along for that, though.